shared/utils/conversion.ts

```diff
@@ -0,0 +1,15 @@
+export function convertScale(rating: number): number;
+export function convertScale(rating: number | undefined): number | undefined;
+export function convertScale(rating: number | undefined) {
+  if (rating === undefined) return undefined;
+  else if (rating < 0) return rating;
+  return rating ** 2;
+}
```

Comment:
ConvertScale is not used with undefined values, so I'd change the definition
```suggestion
export const convertScale = (rating: number) =>
  rating < 0 ? rating : rating ** 2;
```

Reply:
I'd keep the `undefined` check in, since the `$parseDatabaseJson` passes `any` to this.
And for some reason that is called few times with partial data that doesn't have the value.